incubator/mariadb-cluster/templates/slave-configmap.yaml

```diff
@@ -0,0 +1,15 @@
+{{- if and .Values.replication.enabled .Values.slave.config}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "slave.fullname" . }}
+  labels:
+    app: {{ template "name" . }}
+    component: "slave"
+    chart: {{ template "mariadb.chart" . }}
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+data:
+  my.cnf: |-
+{{ .Values.slave.config | indent 4 }}
+{{- end -}}
```

Comment:
I think you don't need the second dash: `{{- end }}`

https://github.com/kubernetes/helm/blob/master/docs/chart_template_guide/control_structures.md

> First, the curly brace syntax of template declarations can be modified with special characters to tell the template engine to chomp whitespace. {{- (with the dash and space added) indicates that whitespace should be chomped left, while -}} means whitespace to the right should be consumed. Be careful! Newlines are whitespace!